shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureSurfaceProducer.java

```diff
@@ -0,0 +1,59 @@
+package io.flutter.embedding.engine.renderer;
+
+import android.graphics.SurfaceTexture;
+import android.view.Surface;
+import androidx.annotation.NonNull;
+import io.flutter.view.TextureRegistry;
+
+/** Uses a {@link android.graphics.SurfaceTexture} to populate the texture registry. */
+final class SurfaceTextureSurfaceProducer
```

Comment:
You need to add a finalizer (see the other impls)

Reply:
Object.finalize() was deprecated in JDK9 and there are plans to remove it. See https://openjdk.org/jeps/421

Reply:
(The JEP points to alternatives with fewer issues)

Reply:
Regardless, we rely on finalizers today so my comment still stands.

Reply:
I filed an issue to keep tabs on these uses: https://github.com/flutter/flutter/issues/141403

Reply:
Done!